presto-main/src/main/java/com/facebook/presto/server/security/PrestoAuthenticator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security;
+
+import com.facebook.airlift.http.server.AuthenticationException;
+import com.facebook.airlift.http.server.Authenticator;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+import java.security.Principal;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoAuthenticator
```

Comment:
Should this be an interface?

Reply:
The `PrestoAuthenticator` class does not need to be an interface because it serves as a concrete adapter between the Airlift `Authenticator`  interface and the pluggable design provided by `PrestoAuthenticator` and `PrestoAuthenticatorFactory` SPI. Its role is to delegate authentication to the `PrestoAuthenticatorManager`.

Reply:
@pramodsatya I understand why you might be confused. We have an interface called PrestoAuthenticator in the presto-spi package with the same name.

@tdcmeehan Currently, we've followed the same naming convention as was done for the PasswordAuthenticator SPI. This could lead to confusion since we have an SPI with the same name as the PasswordAuthenticator class in the presto-main package. Do you think we should refactor the class names in presto-main to eliminate the confusion?

Reply:
Yes.  How about `CustomPrestoAuthenticator`?